src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -251,6 +277,22 @@ static SimplexResult OPT_TryToCallSimplex(const OptimizationOptions& options,
             throw FatalError("Internal error: insufficient memory");
         }
     }
+
+    // WRITE MODELER SOLUTIONS
+    std::stringstream contentStream;
+
+    // we want to only get modeler variables, they're added after legacy vars
+    auto start = solver->variables().begin() + Probleme.NombreDeVariables;
+    for (auto v = start; v < solver->variables().end(); v++)
+    {
+        contentStream << (*v)->name() << "\t" << (*v)->solution_value() << std::endl;
+    }
```

Comment:
Consider using `std::span` (no copy required)
```suggestion
    std::span<MPVariable*> modelerVariables{solver->variables().begin()
                                              + Probleme.NombreDeVariables,
                                            solver->variables().end()};
    for (const auto* v: modelerVariables)
    {
        contentStream << v->name() << "\t" << v->solution_value() << "\n";
    }
```